lib/stronger_parameters/parameters.rb

```diff
@@ -147,7 +147,7 @@ def hash_filter_with_stronger_parameters(params, filter)
         value = fetch(key, nil)
         result = nil
 
-        if value.nil? && self.class.allow_nil_for_everything
+        if key?(key) && value.nil? && self.class.allow_nil_for_everything
```

Comment:
I think what we want is this ?
```
          params[key] = nil if key?(key)
```
... so keep things in sync, if it was not set, don't set it ... but still allow not setting it (skip the value/invalid-value stuff below)